Formula/gimme-aws-creds.rb

```diff
@@ -0,0 +1,75 @@
+class GimmeAwsCreds < Formula
+  include Language::Python::Virtualenv
+
+  desc "CLI to retrieve AWS credentials from Okta"
+  homepage "https://github.com/Nike-Inc/gimme-aws-creds"
+  url "https://github.com/Nike-Inc/gimme-aws-creds/archive/v2.0.2.tar.gz"
+  sha256 "32a28f4d8fc6a72a3a53ab822030f0d00671a79fd4d86823e16464482ee7c664"
+
+  depends_on "python"
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    system libexec/"bin/pip", "install", "-v", "--no-binary", ":all:",
+                              "--ignore-installed", buildpath
+    system libexec/"bin/pip", "uninstall", "-y", "gimme-aws-creds"
```

Comment:
What are these doing and why are they necessary? Thanks!

Reply:
This is modelled on the process that `awscli.rb` uses to install.

Reply:
@Firefishy I'd like to see justification here rather than copy-paste, thanks.

Reply:
@firefishy please justify this thanks.

Reply:
It is used to ensure that no binary pip packages are used in gimme-aws-creds (inc its dependencies) to ensures that no binary dependencies are used see: https://github.com/Homebrew/homebrew-core/pull/5515 
Clean architecture / distro build.

The same build pattern is used by homebrew forumla: autopep8, awscli, awsume, b2-tools, conan, docker-compose, dvc, gimme-aws-creds, khal, notifiers, osc, pipenv, pipx, pre-commit, pwntools, python, python@2, rtv, statik, termius, termtosvg, twarc, vdirsyncer

Reply:
@MikeMcQuaid Per this change in https://github.com/Homebrew/homebrew-core/pull/5500 many other formula use `virtualenv_install_with_resources` which means adding each resource (dependencies) to the formula, instead of trusting the upstream package's `requirements.txt`.

Reply:
From a maintainability I would prefer using upstream `requirements.txt`

Reply:
@Firefishy Agreed. Given that I'm fine with it for now but think it would be ideally some logic in Homebrew/brew to better handle such things. Any ideas on that (won't be on you to implement but you have more context than I do)?